### PR TITLE
kubelet: improve error message about expired certificates

### LINKED ISSUE #98150: kubelet: improve error messages about expired certificates
**What happened**: the `kubelet` fails to start; it logs,

```
Jan 18 17:33:58 aura kubelet[1457370]: E0118 17:33:58.888700 1457370 bootstrap.go:265] part of the existing bootstrap client certificate is expired: 2020-12-06 18:55:55 +0000 UTC
Jan 18 17:33:58 aura kubelet[1457370]: F0118 17:33:58.888743 1457370 server.go:271] failed to run Kubelet: unable to load bootstrap kubeconfig: stat /etc/kubernetes/bootstrap-kubelet.conf: no such file or directory
Jan 18 17:33:58 aura systemd[1]: kubelet.service: Main process exited, code=exited, status=255/EXCEPTION
Jan 18 17:33:58 aura systemd[1]: kubelet.service: Failed with result 'exit-code'.
```

**What you expected to happen**: either the `kubelet` starts, *or it produces a *useable* diagnostic*

It tells me that a certificate is expired. If you Google the error, you'll hit SO questions like [this one](https://stackoverflow.com/questions/59825118/kubelet-unable-to-load-bootstrap-kubeconfig), which helpfully suggest you run `kubeadm alpha certs renew`¹. That runs successfully, but afterwards, the `kubelet` continues to fail to start. A bit more digging, and one might learn about `kubeadm alpha certs check-expiration`, and think, ah-ha! That'll tell me _which_ cert has expired, and thus, where to look for guidance on renewing it.

As we've already run `renew`, it will tell us that …
```
# kubeadm alpha certs check-expiration
CERTIFICATE                EXPIRES                  RESIDUAL TIME   EXTERNALLY MANAGED
admin.conf                 Jan 18, 2022 22:30 UTC   364d            no      
apiserver                  Jan 18, 2022 22:30 UTC   364d            no      
apiserver-etcd-client      Jan 18, 2022 22:30 UTC   364d            no      
apiserver-kubelet-client   Jan 18, 2022 22:30 UTC   364d            no      
controller-manager.conf    Jan 18, 2022 22:30 UTC   364d            no      
etcd-healthcheck-client    Jan 18, 2022 22:30 UTC   364d            no      
etcd-peer                  Jan 18, 2022 22:30 UTC   364d            no      
etcd-server                Jan 18, 2022 22:30 UTC   364d            no      
front-proxy-client         Jan 18, 2022 22:30 UTC   364d            no      
scheduler.conf             Jan 18, 2022 22:30 UTC   364d            no      
```
…nothing is expired. (But still the `kubelet` refuses to start.)

So we just run every `*.crt` in `/etc/kubernetes` through `openssl x509` manually with `find`:

```
# find -iname '*.crt' -print0 | xargs -0 -n1 openssl x509 -noout -text -in | grep Not\ After
```

…and they're all fine.

After some further digging, I did find it. It's in `kubelet.conf`, but it's wrapped in YAML config so the `find` can't see it. It isn't clear to me why `kubeadm` should miss it. But at the very least, **`kubelet` needs to emit a usable diagnostic**, minimally including *which* certificate is expired, and ideally with enough information for me to determine the correct way to fix it.

(Eventually, I hit [this comment on a closed issue](https://github.com/kubernetes/kubernetes/issues/84252#issuecomment-545622426), which contains the additional command to regenerate `kubelet.conf`.)

**How to reproduce it (as minimally and precisely as possible)**: run a cluster for ~1 yr, let the cert expire…

**Anything else we need to know?**:

¹it instills much confidence that renewing the certs is "alpha".

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16.8
- Cloud provider or hardware configuration: N/A, single `kubeadm`-made node
- OS (e.g: `cat /etc/os-release`): Gentoo
- Kernel (e.g. `uname -a`): `5.4.60-gentoo`
- Install tools: `kubeadm`?
- Network plugin and version (if this is a network-related bug): N/A, but Calico
- Others:

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
kubelet: improve error message about expired certificates.
Print the expired certificate file path in the error message.

**Which issue(s) this PR fixes**:

Fixes #98150

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func isClientConfigStillValid(kubeconfigPath string) (bool, error) {
 	now := time.Now()
 	for _, cert := range certs {
 		if now.After(cert.NotAfter) {
-			utilruntime.HandleError(fmt.Errorf("part of the existing bootstrap client certificate is expired: %s", cert.NotAfter))
+			utilruntime.HandleError(fmt.Errorf("part of the existing bootstrap client certificate which located at %q is expired: %s", kubeconfigPath, cert.NotAfter))
```
Comment (neolit123) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go:265]: SGTM, suggesting a minor change:

```suggestion
			utilruntime.HandleError(fmt.Errorf("part of the existing bootstrap client certificate located at %q is expired: %v", kubeconfigPath, cert.NotAfter))
```

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by neolit123** [comment]
bazel flake, maybe
/retest

**Review by ehashman — APPROVED** [review]
/lgtm

**Review by pacoxu — APPROVED** [review]

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 2]
```diff
@@ -262,7 +262,7 @@ func isClientConfigStillValid(kubeconfigPath string) (bool, error) {
 	now := time.Now()
 	for _, cert := range certs {
 		if now.After(cert.NotAfter) {
-			utilruntime.HandleError(fmt.Errorf("part of the existing bootstrap client certificate is expired: %s", cert.NotAfter))
+			utilruntime.HandleError(fmt.Errorf("part of the existing bootstrap client certificate located at %q is expired: %v", kubeconfigPath, cert.NotAfter))
```
Comment (liggitt) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go:265]: should we add the path to all the error messages returned from this function?

Comment (SataQiu) [reply pkg/kubelet/certificate/bootstrap/bootstrap.go]: make sense

**Review by ehashman — APPROVED** [review]
awesome!

/lgtm

**Pull request merged**